spec/models/mixins/relationship_mixin_spec.rb

```diff
@@ -356,6 +356,12 @@ def self.name
         service2.save!
         expect(service6.with_relationship_type("custom") { service6.child_ids }).to eq([service2.id])
       end
+
+      it "#replace_children" do
+        service7 = ancestry_class.create
+        service1.with_relationship_type("custom") { service1.replace_children(service7, service4) }
+        expect(service1.with_relationship_type("custom") { service1.reload.children }).to contain_exactly(service7, service4)
+      end
```

Comment:
@d-m-u could we cover a few more possible cases like:
1. replace_children() with no children removes everything
2. replace_children() with the same set of current children doesn't change anything
3. replace_children() with no children before adds only new children (I assume the current test covers this case)
4. replace_children() with the same children except 1 deletes that one child
5. replace_children() with the same children plus 1 extra adds that one child

Reply:
the current test covers 5, just fyi

Reply:
:+1: thanks I wasn't sure what children service1 already had in this spec